TYPO3.Flow/Classes/TYPO3/Flow/Composer/Utility.php

```diff
@@ -0,0 +1,176 @@
+<?php
+namespace TYPO3\Flow\Composer;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow framework.                       *
+ *                                                                        *
+ * It is free software; you can redistribute it and/or modify it under    *
+ * the terms of the GNU Lesser General Public License, either version 3   *
+ * of the License, or (at your option) any later version.                 *
+ *                                                                        *
+ * The TYPO3 project - inspiring people to share!                         *
+ *                                                                        */
+
+use TYPO3\Flow\Package\PackageInterface;
+use TYPO3\Flow\Reflection\ObjectAccess;
+use TYPO3\Flow\Utility\Files;
+
+/**
+ * Utility to access composer information like composer manifests (composer.json) and the lock file.
+ */
+class Utility
+{
+    /**
+     * Runtime cache for composer.json data
+     *
+     * @var array
+     */
+    protected static $composerManifestCache;
+
+    /**
+     * Runtime cache for composer.lock data
+     *
+     * @var array
+     */
+    protected static $composerLockCache;
+
+    /**
+     * Returns contents of Composer manifest - or part there of.
+     *
+     * @param string $manifestPath
+     * @param string $configurationPath Optional. Only return the part of the manifest indexed by configurationPath
+     * @return mixed
+     * @see json_decode for return values
+     */
+    public static function getComposerManifest($manifestPath, $configurationPath = null)
+    {
+        $composerManifest = static::readComposerManifest($manifestPath);
+        if ($composerManifest === null) {
+            return null;
+        }
+
+        if ($configurationPath !== null) {
+            return ObjectAccess::getPropertyPath($composerManifest, $configurationPath);
+        } else {
+            return $composerManifest;
+        }
+    }
+
+    /**
+     * Read the content of the composer.lock
+     *
+     * @return array
+     */
+    public static function readComposerLock()
+    {
+        if (self::$composerLockCache === null) {
+            if (!file_exists(FLOW_PATH_ROOT . 'composer.lock')) {
+                return array();
```

Comment:
return []; would be more consistent with the further boyscouting in the change ;)